File: CountryWiki.Angular/src/app/mappers/countryReplyMapper.ts
import { CountryReply } from "../generated/country.shared_pb";
import { CountryModel } from "../models/countryModel";

export class CountryReplyMapper {
    public static Map(country: CountryModel, countryReply: CountryReply.AsObject) {
        if (countryReply === null || countryReply === null)
        return;
    
        country.id = countryReply.id;
        country.name = countryReply.name;
        country.description = countryReply.description;
        country.capitalCity = countryReply.capitalcity;
        country.flagUri = countryReply.flaguri;
        country.anthem = countryReply.anthem;
        country.languages = countryReply.languagesList;
    }
}

File: CountryService/Mappers/CountryReplyMapper.cs
namespace CountryService.gRPC.Mappers;

public static class CountryReplyMapper
{
    public static CountryReply ToReply(this CountryModel country)
    {
        if (country is null)
            return null;

        var countryReply = new CountryReply
        {
            Id = country.Id,
            Name = country.Name,
            Description = country.Description,
            Anthem = country.Anthem,
            CapitalCity = country.CapitalCity,
            FlagUri = country.FlagUri
        };
        countryReply.Languages.AddRange(country.Languages);

        return countryReply;
    }
}